src/listeners.ts

```diff
@@ -0,0 +1,582 @@
+        },
+    );
+
+    // ================= Uncategorized =================
+
+    $(document).on(
+        "keypress" + ULABEL_NAMESPACE,
+        function (keypress_event: JQuery.KeyPressEvent) {
+            handle_keypress_event(keypress_event, ulabel);
+        },
+    );
+
+    // This listener does not use jquery because it requires being able to prevent default
+    // There are maybe some hacky ways to do this with jquery
+    // https://stackoverflow.com/questions/60357083/does-not-use-passive-listeners-to-improve-scrolling-performance-lighthouse-repo
+    // Detection ctrl+scroll
+    document.getElementById(
+        ulabel.config["annbox_id"],
+    ).addEventListener(
+        "wheel",
+        ulabel.handle_wheel.bind(ulabel),
+    );
+
+    // Create a resize observer to reposition dialogs
+    const dialog_resize_observer = new ResizeObserver(function () {
+        ulabel.reposition_dialogs();
+    });
+
+    // Observe the changes on the imwrap_id element
+    dialog_resize_observer.observe(
+        document.getElementById(ulabel.config["imwrap_id"]),
+    );
+
+    // Store a reference
+    ulabel.resize_observers.push(dialog_resize_observer);
+
+    // Create a resize observer to handle toolbox overflow
+    const tb_overflow_resize_observer = new ResizeObserver(function () {
+        ulabel.handle_toolbox_overflow();
+    });
+
+    // Observe the changes on the ulabel container
+    tb_overflow_resize_observer.observe(
+        document.getElementById(ulabel.config["container_id"]),
+    );
+
+    // Store a reference
+    ulabel.resize_observers.push(tb_overflow_resize_observer);
+
+    // create_soft_id_toolbox_button_listener(ulabel);
+    $(document).on(
+        "click" + ULABEL_NAMESPACE,
+        `#${ulabel.config["toolbox_id"]} a.tbid-opt`,
+        function (click_event: JQuery.ClickEvent) {
+            handle_soft_id_toolbox_button_click(click_event, ulabel);
+        },
+    );
+
+    $(document).on(
+        "click" + ULABEL_NAMESPACE,
+        "a.tb-st-switch[href]",
+        function (click_event) {
+            const switch_to = $(click_event.target).attr("id").split("--")[1];
+
+            // Ignore if in the middle of annotation
+            if (ulabel.get_current_subtask()["state"]["is_in_progress"]) return;
+
+            ulabel.set_subtask(switch_to);
+        },
+    );
+
+    // Keybind to switch active subtask
+    $(document).on(
+        "keypress" + ULABEL_NAMESPACE,
+        function (keypress_event) {
+            // Ignore if in the middle of annotation
+            if (ulabel.get_current_subtask()["state"]["is_in_progress"]) return;
+
+            // Check for the right keypress
+            if (keypress_event.key === ulabel.config.switch_subtask_keybind) {
+                ulabel.switch_to_next_subtask();
+            }
+        },
+    );
+
+    $(document).on(
+        "input" + ULABEL_NAMESPACE,
+        "input.frame_input",
+        () => ulabel.update_frame(),
```

Comment:
I left some of these one-liners as anonymous arrow functions as this is plenty readable. 
I think these could also be reduced, e.g. to just `ulabel.update_frame`

Reply:
Well, it would need to be `ulabel.update_frame.bind(ulabel)`, which isn't really more readable. Other comment thread will sort this out.